pkg/operation/shootsecrets/secrets_manager.go

```diff
@@ -1,158 +0,0 @@
-	DeployedSecrets          map[string]*corev1.Secret
-}
-
-// NewSecretsManager takes in a list of GardenerResourceData items, a static token secret config, a map of certificate authority configs,
-// a function which can generate secret configurations and returns a new SecretsManager struct
-func NewSecretsManager(
-	gardenerResourceDataList gardencorev1alpha1helper.GardenerResourceDataList,
-	secretConfigGenerator SecretConfigGeneratorFunc,
-) *SecretsManager {
-	return &SecretsManager{
-		GardenerResourceDataList: gardenerResourceDataList,
-		secretConfigGenerator:    secretConfigGenerator,
-		certificateAuthorities:   make(map[string]*secrets.Certificate),
-		existingSecrets:          map[string]*corev1.Secret{},
-		DeployedSecrets:          map[string]*corev1.Secret{},
-	}
-}
-
-// WithExistingSecrets adds the provided map of existing secrets to the SecretsManager
-func (s *SecretsManager) WithExistingSecrets(existingSecrets map[string]*corev1.Secret) *SecretsManager {
-	s.existingSecrets = existingSecrets
-	return s
-}
-
-// WithCertificateAuthorities adds the provided map of CA secrets to the SecretsManager
-func (s *SecretsManager) WithCertificateAuthorities(cas map[string]*secrets.Certificate) *SecretsManager {
-	s.certificateAuthorities = cas
-	return s
-}
-
-// Generate generates InfoData for all shoot secrets managed by gardener and adds it to the SecretManager's
-// GardenerResourceData
-func (s *SecretsManager) Generate() error {
-	secretConfigs, err := s.secretConfigGenerator(s.certificateAuthorities)
-	if err != nil {
-		return err
-	}
-
-	for _, config := range secretConfigs {
-		if err := s.generateInfoDataAndUpdateResourceList(config); err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
-// Deploy gets InfoData for all shoot secrets managed by gardener from the SecretManager's GardenerResourceDataList
-// and uses it to generate kubernetes secrets and deploy them in the provided namespace.
-func (s *SecretsManager) Deploy(ctx context.Context, k8sClient client.Client, namespace string) error {
-	if s.secretConfigGenerator == nil {
-		return nil
-	}
-
-	secretConfigs, err := s.secretConfigGenerator(s.certificateAuthorities)
-	if err != nil {
-		return err
-	}
-
-	deployedSecrets, err := secrets.GenerateClusterSecretsWithFunc(ctx, k8sClient, s.existingSecrets, secretConfigs, namespace, func(c secrets.ConfigInterface) (secrets.DataInterface, error) {
-		return s.getInfoDataAndGenerateSecret(c)
-	})
-	if err != nil {
-		return err
-	}
-
-	for name, secret := range deployedSecrets {
-		s.DeployedSecrets[name] = secret
-	}
-
-	return nil
-}
-
-func (s *SecretsManager) generateInfoDataAndUpdateResourceList(secretConfig secrets.ConfigInterface) error {
-	if s.GardenerResourceDataList.Get(secretConfig.GetName()) != nil {
-		return nil
-	}
-	data, err := secretConfig.GenerateInfoData()
-	if err != nil {
-		return err
-	}
-	return infodata.UpsertInfoData(&s.GardenerResourceDataList, secretConfig.GetName(), data)
```

Comment:
Can we drop this whole info data stuff now?

Reply:
Yeah 🎉 Thanks for reminding. I will amend this to the cleanup commit. That feels good :)